src/OpenConext/ProfileBundle/Controller/MyServicesController.php

```diff
@@ -31,38 +33,42 @@ class MyServicesController
     private $templateEngine;
 
     /**
-     * @var TokenStorageInterface
+     * @var SpecifiedConsentListService
      */
-    private $tokenStorage;
+    private $specifiedConsentListService;
 
     /**
-     * @var ConsentService
+     * @var UserProvider
      */
-    private $consentService;
+    private $userProvider;
 
     /**
      * @param EngineInterface $templateEngine
-     * @param TokenStorageInterface $tokenStorage
-     * @param ConsentService $consentService
+     * @param UserProvider $userProvider
+     * @param SpecifiedConsentListService $specifiedConsentListService
      */
     public function __construct(
         EngineInterface $templateEngine,
-        TokenStorageInterface $tokenStorage,
-        ConsentService $consentService
+        UserProvider $userProvider,
+        SpecifiedConsentListService $specifiedConsentListService
     ) {
-        $this->templateEngine = $templateEngine;
-        $this->tokenStorage   = $tokenStorage;
-        $this->consentService = $consentService;
+        $this->templateEngine              = $templateEngine;
+        $this->userProvider                = $userProvider;
+        $this->specifiedConsentListService = $specifiedConsentListService;
     }
 
     public function overviewAction()
     {
-        $user        = $this->tokenStorage->getToken()->getUser();
-        $consentList = $this->consentService->findAllFor($user);
+        if (!$this->userProvider->hasCurrentUser()) {
+            throw new RuntimeException('Cannot retrieve MyServices overview: no user available');
```

Comment:
It felt weird when I put it there. Indeed, the whole application (except of course logging in/saml and the public metadata) assumes someone to be logged in so it should be enforced at that level, but I felt some defense was needed in any case (so misconfiguration or misuse of components will be easily detected). 

Will make it a todo for now so we can pick it up later.